subgraphs/_reference_/src/sdk/protocols/perpfutures/pool.ts

```diff
@@ -0,0 +1,810 @@
+   * of type LIQUIDITY. Or pools that emit some kind of LP token on deposit.
+   * It will also update the outputTokenPriceUSD value.
+   * @param amount
+   */
+  setOutputTokenSupply(amount: BigInt): void {
+    this.pool.outputTokenSupply = amount;
+    this.refreshOutputTokenPriceUSD();
+    this.save();
+  }
+
+  /**
+   * Updates the price of the pool's output token in USD.
+   * This is automatically called when changing the output token supply via setOutputTokenSupply
+   * but can be called manually if necessary.
+   */
+  refreshOutputTokenPriceUSD(): void {
+    if (!this.pool.outputToken) return;
+
+    const token = this.tokens.getOrCreateTokenFromBytes(this.pool.outputToken);
+    const price = this.protocol.pricer.getTokenPrice(token);
+
+    this.pool.outputTokenPriceUSD = price;
+    this.save();
+  }
+
+  /**
+   * Adds a given amount to the pool's stakedOutputTokenAmount.
+   * @param amount
+   * @returns
+   */
+  addStakedOutputTokenAmount(amount: BigInt): void {
+    if (!this.pool.stakedOutputTokenAmount)
+      this.pool.stakedOutputTokenAmount = constants.BIGINT_ONE;
+
+    this.pool.stakedOutputTokenAmount =
+      this.pool.stakedOutputTokenAmount!.plus(amount);
+    this.save();
+  }
+
+  /**
+   * Sets the pool's stakedOutputTokenAmount value.
+   * @param amount
+   */
+  setStakedOutputTokenAmount(amount: BigInt): void {
+    this.pool.stakedOutputTokenAmount = amount;
+    this.save();
+  }
+
+  /**
+   * Utility function to convert some amount of input token to USD.
+   *
+   * @param amount the amount of inputTokens to convert to USD
+   * @returns The converted amount.
+   */
+  getInputTokenAmountPrice(token: TokenSchema, amount: BigInt): BigDecimal {
+    const price = this.protocol.getTokenPricer().getTokenPrice(token);
+    token.lastPriceUSD = price;
+    token.save();
+
+    return amount.divDecimal(exponentToBigDecimal(token.decimals)).times(price);
+  }
+
+  /**
+   * Sets the pool's input token balances to the given amount. It will optionally
+   * update the pool's and protocol's total value locked. If not stated, will default to true.
+   *
+   * @param amount amount to be set as the pool's input token balance.
+   * @param updateMetrics optional parameter to indicate whether to update the pool's and protocol's total value locked.
+   */
+  setInputTokenBalances(
+    newBalances: [BigInt],
+    updateMetrics: boolean = true
+  ): void {
+    this.pool.inputTokenBalances = newBalances;
+    this.save();
+
+    if (updateMetrics) this.refreshTotalValueLocked();
+  }
+
+  /**
+   * Sets the rewardTokenEmissions and its USD value for a given reward token.
+   * It will also create the RewardToken entity and add it to the pool rewardTokens array
+   * if not already present.
+   * @param type The type of reward token
+   * @param token The actual token being rewarded
+   * @param amount The daily amount of reward tokens emitted to this pool
+   */
+  setRewardEmissions(
```

Comment:
what if there is more than 1 reward token?

Reply:
This function can be used within a for loop when calculating the daily reward emission amount by the user.